jscomp/core/lam_compile.ml

```diff
@@ -187,12 +187,24 @@ type initialization = J.block
    -: we should not do functor application inlining in a
       non-toplevel, it will explode code very quickly
 *)
-let rec compile_external_field (* Like [List.empty]*)
+let rec compile_external_field (* Like [List.empty] *)
     (lamba_cxt : Lam_compile_context.t) (id : Ident.t) name : Js_output.t =
+  (* NOTE(anmonteiro): This function checks whether there's inlining
+     information available for the lambda being compiled.
+
+     The fallback case happens in 2 scenarios:
+
+       1. there's no inlining information available in the `.cmj` file
+       2. there's no `.cmj` file available. This can happen if we're compiling
+          a dune virtual library where one of the modules uses a binding from
+          any of its virtual modules. Because we're programming against the
+          interface file at this point, we can emit the fallback expression
+          too.
+  *)
   match Lam_compile_env.query_external_id_info id name with
   | { persistent_closed_lambda = Some lam; _ } when Lam_util.not_function lam ->
       compile_lambda lamba_cxt lam
-  | _ ->
+  | _ | (exception Mel_exception.Error (Cmj_not_found _)) ->
```

Comment:
Probably out of scope, but I wonder if it'd make sense to have `load_unit` return a result type or some other variant, rather than raise exceptions? 

I can see other usages of that function and `query_external_id_info` that don't process `Cmj_not_found`, and after the changes to support virtual libraries, that's not an unexpected situation anymore.

Reply:
I feel like we want it to fail when a `.cmj` isn't found, except for this one time where we know it could not be there.

In principle, all the other usages should have a `.cmj` file present that we can lean on.

Reply:
Oh you might actually be right. I think I missed a few usages